lib/spack/spack/solver/asp.py

```diff
@@ -2488,6 +2488,13 @@ def _reusable_specs(self):
                 # TODO: update mirror configuration so it can indicate that the
                 # TODO: source cache (or any mirror really) doesn't have binaries.
                 pass
+
+        # If we only want to reuse dependencies, remove the root specs
+        if self.reuse == "dependencies":
```

Comment:
(Comment, not request): If we add more reuse modes, it is worth turning these literal strings into `enums`.